### PR TITLE
8329617: Update stylesheet for specs and tool documentation

### PR DESCRIPTION
**Pull request by hns**
Please review an update to the `jdk-default.css` stylesheet used for specifications and tool guides. The original purpose was to make use of the Dejavu web fonts provided by the API docs and to update the navigation bar to match the one in the API docs. However, the updates include some other fixes and improvements also described below.

 - The change to use the DejaVu web fonts consists only of the `@import` statement in line 16 as the stylesheet already used DejaVu web fonts as first choice in its `font-family` rules.
 - The changes to make the navigation bar match the one in the API docs are mostly located at the end of the file (beyond line 160). However, this also includes setting the `margin` property to '0' in the `body` element and adding a `margin` in the `main` and `footer` elements instead. 
 - To set the horizontal margin for page content elements outside the `main` element which occur in some pages, a margin is set explicitly on those elements in lines 48-50. While this is a bit awkward, I think it's still better than working with negative margins in the header to offset the margin in the `body` element.
 - Most of the remaining changes (lines 53-110) are changes are to redefine the styles in simpler terms, such as leaving out declarations that are equal to browser defaults, and removing the units from `0`-length values.

The changes are intended to preserve the layout of the pages, including the body font size which is slightly different from the one used in API docs (`10pt` vs `14px`). I can provide before/after snapshots of the rendered documentation if desired.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8329617](https://bugs.openjdk.org/browse/JDK-8329617): Update stylesheet for specs and tool documentation (**Enhancement** - P4)

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18694/head:pull/18694` \
`$ git checkout pull/18694`

Update a local copy of the PR: \
`$ git checkout pull/18694` \
`$ git pull https://git.openjdk.org/jdk.git pull/18694/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18694`

View PR using the GUI difftool: \
`$ git pr show -t 18694`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18694.diff">https://git.openjdk.org/jdk/pull/18694.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18694#issuecomment-2045154435)

### CODE REVIEW

**Comment by jonathan-gibbons** [comment]
Is there a reason to want the main font size to be different between the API pages and spec pages?

FWIW, while the PR description says the body font size is 10pt vs 14px, on representative text for the `jdk.javadoc` module and Doc Comment Spec, I observed 13.333px vs 14px.

**Comment by hns** [comment]
> Is there a reason to want the main font size to be different between the API pages and spec pages?

I did not want to change the font size in this patch. I started by changing the font size to 14 px, but went back to the initial 10 pt in order to detect any unintentional changes. To me the specs/man pages look good with the original font size, so I kept it that way. We can increase it if we think it's too small.

> FWIW, while the PR description says the body font size is 10pt vs 14px, on representative text for the `jdk.javadoc` module and Doc Comment Spec, I observed 13.333px vs 14px.

That's expected, as [10 pt corresponds to 13.33 px](https://cssunitconverter.com/pt-to-px/).

**Review by jonathan-gibbons — APPROVED** [review]
@hns
I'll approve the work as-is, but would encourage you to consider the possibility of additional cleanup to further align the styles between the api pages and specs pages.
